From 1e5b70549566a6e108d661d6ec48f9318ec05aff Mon Sep 17 00:00:00 2001
From: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Date: Mon, 4 Nov 2024 14:29:37 -0300
Subject: [PATCH] wrapper: Automatically enable --no-sandbox for root user

Running as root without --no-sandbox is not supported, so let's
automatically set that option with a warning message.

Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
---
 chrome/installer/linux/common/wrapper | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/chrome/installer/linux/common/wrapper b/chrome/installer/linux/common/wrapper
index 7d8c8dd5fb9c8..cf89bb168ed8d 100755
--- a/chrome/installer/linux/common/wrapper
+++ b/chrome/installer/linux/common/wrapper
@@ -38,5 +38,11 @@ exec 2> >(exec cat >&2)

 CHROME_EXTRA_ARGS=""

+if [ $(id -u) -eq 0 ]; then
+  echo "WARNING: Running as root without --no-sandbox is not supported. See https://crbug.com/638180."
+  echo "WARNING: Enabling --no-sandbox runtime parameter."
+  CHROME_EXTRA_ARGS="${CHROME_EXTRA_ARGS} --no-sandbox "
+fi
+
 # Note: exec -a below is a bashism.
 exec -a "$0" "$HERE/@@PROGNAME@@" ${CHROME_EXTRA_ARGS} "$@"
--
2.45.2

